src/InputToolbar.tsx

```diff
@@ -47,128 +47,96 @@ export interface InputToolbarProps<TMessage extends IMessage> {
-export default class InputToolbar<
-  TMessage extends IMessage = IMessage
-> extends React.Component<InputToolbarProps<TMessage>, { position: string }> {
-  static defaultProps = {
-    renderAccessory: null,
-    renderActions: null,
-    renderSend: null,
-    renderComposer: null,
-    containerStyle: {},
-    primaryStyle: {},
-    accessoryStyle: {},
-    onPressActionButton: () => {},
-  }
-
-  static propTypes = {
-    renderAccessory: PropTypes.func,
-    renderActions: PropTypes.func,
-    renderSend: PropTypes.func,
-    renderComposer: PropTypes.func,
-    onPressActionButton: PropTypes.func,
-    containerStyle: StylePropType,
-    primaryStyle: StylePropType,
-    accessoryStyle: StylePropType,
-  }
-
-  state = {
-    position: 'absolute',
-  }
-
-  keyboardWillShowListener?: EmitterSubscription = undefined
-  keyboardWillHideListener?: EmitterSubscription = undefined
-
-  componentDidMount() {
-    this.keyboardWillShowListener = Keyboard.addListener(
-      'keyboardWillShow',
-      this.keyboardWillShow,
-    )
-    this.keyboardWillHideListener = Keyboard.addListener(
-      'keyboardWillHide',
-      this.keyboardWillHide,
-    )
-  }
-
-  componentWillUnmount() {
-    if (this.keyboardWillShowListener) {
-      this.keyboardWillShowListener.remove()
+export function InputToolbar<TMessage extends IMessage = IMessage>(props: InputToolbarProps<TMessage>) {
+  const [position, setPosition] = useState('absolute');
+
+  useEffect(() => {
+    const keyboardWillShowListener = Keyboard.addListener('keyboardWillShow', () => setPosition('relative'));
+    return () => keyboardWillShowListener?.remove();
+  }, []);
+
+  useEffect(() => {
+    const keyboardWillHideListener = Keyboard.addListener('keyboardWillHide', () => setPosition('absolute'));
+    return () => keyboardWillHideListener?.remove();
+  }, []);
+
+  const renderActions = useCallbackOne(() => {
+    const { containerStyle, ...rest } = props
+    if (props.renderActions) {
+      return props.renderActions(rest)
+    } else if (props.onPressActionButton) {
+      return <Actions {...rest} />
     }
-    if (this.keyboardWillHideListener) {
-      this.keyboardWillHideListener.remove()
-    }
-  }
-
-  keyboardWillShow = () => {
-    if (this.state.position !== 'relative') {
-      this.setState({
-        position: 'relative',
-      })
-    }
-  }
 
-  keyboardWillHide = () => {
-    if (this.state.position !== 'absolute') {
-      this.setState({
-        position: 'absolute',
-      })
-    }
-  }
-
-  renderActions() {
-    const { containerStyle, ...props } = this.props
-    if (this.props.renderActions) {
-      return this.props.renderActions(props)
-    } else if (this.props.onPressActionButton) {
-      return <Actions {...props} />
-    }
     return null
-  }
+  }, [props.renderActions, props.onPressActionButton]);
 
-  renderSend() {
-    if (this.props.renderSend) {
-      return this.props.renderSend(this.props)
+  const renderSend = useCallbackOne(() => {
```

Comment:
The same message concerning callback returning a component, please move under return.